packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue

```diff
@@ -1000,56 +1000,46 @@ export default mixins(showMessage, nodeHelpers).extend({
-				if (credentialsForType) {
-					this.selectedCredential = credentialsForType.name;
-					if (this.credentialType) {
-						for (const property of this.credentialType.properties) {
-							if (
-								!this.credentialData.hasOwnProperty(property.name) &&
-								!this.credentialType.__overwrittenProperties?.includes(property.name)
-							) {
-								Vue.set(
-									this.credentialData,
-									property.name,
-									property.default as CredentialInformation,
-								);
-							}
-						}
-					}
-					this.nodeAccess = this.nodesWithAccess.reduce(
-						(accu: NodeAccessMap, node: { name: string }) => {
-							if (this.mode === 'new') {
-								accu[node.name] = { nodeType: node.name }; // enable all nodes by default
-							} else {
-								accu[node.name] = null;
-							}
-
-							return accu;
-						},
-						{},
-					);
-					// Update current node auth type so credentials dropdown can be displayed properly
-					updateNodeAuthType(this.ndvStore.activeNode, type);
-					// Also update credential name but only if the default name is still used
-					if (this.hasUnsavedChanges && !this.hasUserSpecifiedName) {
-						const newDefaultName = await this.credentialsStore.getNewCredentialName({
-							credentialTypeName: this.getDefaultCredentialTypeName(),
-						});
-						this.credentialName = newDefaultName;
-					}
+			if (!this.activeNodeType?.credentials) {
+				return;
+			}
+			const credentialsForType = getNodeCredentialForSelectedAuthType(this.activeNodeType, type);
+			if (credentialsForType) {
+				this.selectedCredential = credentialsForType.name;
+				this.resetCredentialData();
+				this.setupNodeAccess();
+				// Update current node auth type so credentials dropdown can be displayed properly
+				updateNodeAuthType(this.ndvStore.activeNode, type);
+				// Also update credential name but only if the default name is still used
+				if (this.hasUnsavedChanges && !this.hasUserSpecifiedName) {
+					const newDefaultName = await this.credentialsStore.getNewCredentialName({
+						credentialTypeName: this.getDefaultCredentialTypeName,
+					});
+					this.credentialName = newDefaultName;
 				}
 			}
 		},
-		getDefaultCredentialTypeName(): string {
-			let credentialTypeName = this.credentialTypeName;
-			if (!credentialTypeName || credentialTypeName === 'null') {
-				if (this.activeNodeType && this.activeNodeType.credentials) {
-					credentialTypeName = this.activeNodeType.credentials[0].name;
-				}
+		setupNodeAccess(): void {
+			this.nodeAccess = this.nodesWithAccess.reduce(
+				(accu: NodeAccessMap, node: { name: string }) => {
+					if (this.mode === 'new') {
+						accu[node.name] = { nodeType: node.name }; // enable all nodes by default
+					} else {
+						accu[node.name] = null;
+					}
+
+					return accu;
+				},
+				{},
+			);
+		},
+		resetCredentialData(): void {
+			if (!this.credentialType) {
+				return;
+			}
+			for (const property of this.credentialType.properties) {
```

Comment:
should this check for overrides still as original logic does? as not to break cloud behavior for example...

Reply:
please make sure this works on cloud with overrides

Reply:
Yes, adding it back.